Formula/p/postgresql@17.rb

```diff
@@ -0,0 +1,178 @@
+                                    "includedir=#{include}/postgresql"
+
+    # Modify the Makefile again to now use opt_lib path rather than Cellar path for dependents.
+    inreplace lib/"postgresql/pgxs/src/Makefile.shlib", "-install_name '#{lib}/", "-install_name '#{opt_lib}/"
+  end
+
+  def post_install
+    (var/"log").mkpath
+    postgresql_datadir.mkpath
+
+    # Manually link files from keg to non-conflicting versioned directories in HOMEBREW_PREFIX.
+    %w[include lib share].each do |dir|
+      dst_dir = HOMEBREW_PREFIX/dir/name
+      src_dir = prefix/dir/"postgresql"
+      src_dir.find do |src|
+        dst = dst_dir/src.relative_path_from(src_dir)
+
+        # Retain existing real directories for extensions if directory structure matches
+        next if dst.directory? && !dst.symlink? && src.directory? && !src.symlink?
+
+        rm_r(dst) if dst.exist? || dst.symlink?
+        if src.symlink? || src.file?
+          Find.prune if src.basename.to_s == ".DS_Store"
+          dst.parent.install_symlink src
+        elsif src.directory?
+          dst.mkpath
+        end
+      end
+    end
+
+    # Also link versioned executables
+    bin.each_child { |f| (HOMEBREW_PREFIX/"bin").install_symlink f => "#{f.basename}-#{version.major}" }
+
+    # Don't initialize database, it clashes when testing other PostgreSQL versions.
+    return if ENV["HOMEBREW_GITHUB_ACTIONS"]
+
+    system bin/"initdb", "--locale=C", "-E", "UTF-8", postgresql_datadir unless pg_version_exists?
+  end
+
+  def postgresql_datadir
+    var/name
+  end
+
+  def postgresql_log_path
+    var/"log/#{name}.log"
+  end
+
+  def pg_version_exists?
+    (postgresql_datadir/"PG_VERSION").exist?
+  end
+
+  def caveats
+    <<~EOS
+      This formula has created a default database cluster with:
+        initdb --locale=C -E UTF-8 #{postgresql_datadir}
+      For more details, read:
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
+    EOS
+  end
+
+  service do
+    run [opt_bin/"postgres", "-D", f.postgresql_datadir]
+    environment_variables LC_ALL: "C"
+    keep_alive true
+    log_path f.postgresql_log_path
+    error_log_path f.postgresql_log_path
+    working_dir HOMEBREW_PREFIX
+  end
+
+  test do
+    system "#{bin}/initdb", testpath/"test" unless ENV["HOMEBREW_GITHUB_ACTIONS"]
+    assert_equal "#{HOMEBREW_PREFIX}/share/#{name}", shell_output("#{bin}/pg_config --sharedir").chomp
+    assert_equal "#{HOMEBREW_PREFIX}/lib/#{name}", shell_output("#{bin}/pg_config --libdir").chomp
+    assert_equal "#{HOMEBREW_PREFIX}/lib/#{name}", shell_output("#{bin}/pg_config --pkglibdir").chomp
+    assert_equal "#{HOMEBREW_PREFIX}/include/#{name}", shell_output("#{bin}/pg_config --pkgincludedir").chomp
+    assert_equal "#{HOMEBREW_PREFIX}/include/#{name}/server",
+                 shell_output("#{bin}/pg_config --includedir-server").chomp
+    assert_match "-I#{Formula["gettext"].opt_include}", shell_output("#{bin}/pg_config --cppflags")
```

Comment:
Linux is probably going to fail here as I removed `gettext` dep. Shouldn't be needed based on original reason (libintl.h) as that is in glibc.